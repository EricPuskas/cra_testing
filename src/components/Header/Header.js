import React from "react";
import "./Header.scss";
import Logo from "../../assets/logo.png";

const Header = () => {
  return (
    <header data-test="header">
      <div className="wrap">
        <div className="logo">
          <img src={Logo} alt="logo" data-test="logo" />
        </div>
        <div className="logo-title">CRA-Testing</div>
      </div>
    </header>
  );
};

export default Header;
