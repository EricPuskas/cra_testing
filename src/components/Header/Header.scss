header {
  display: block;
  width: 100%;
  margin: 0 auto;
  padding: 0 10px;
  height: 50px;
  background: #121212;
  .wrap {
    position: relative;
    display: block;
    width: 100%;
    margin: 0 auto;
    height: 100%;
    .logo {
      display: block;
      width: 50px;
      position: absolute;
      left: 10px;
      top: 50%;
      transform: translateY(-50%);
      -webkit-transform: translateY(-50%);
      -moz-transform: translateY(-50%);
      img {
        display: block;
        margin: 0;
        width: 100%;
      }
    }
    .logo-title {
      align-items: center;
      margin-left: 65px;
      display: flex;
      font-size: 1.3rem;
      color: #fff;
      height: 50px;
    }
  }
}
