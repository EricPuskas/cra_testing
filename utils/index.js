import checkPropTypes from "check-prop-types";
import { applyMiddleware, createStore } from "redux";
import rootReducer from "../src/reducers";
import { middleware } from "../src/store";

export const findByAttribute = (component, attr) =>
  component.find(`[data-test="${attr}"]`);

export const checkProps = (component, expectedProps) =>
  checkPropTypes(component.propTypes, expectedProps, "props", component.name);

export const testStore = initState => {
  const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
  return createStoreWithMiddleware(rootReducer, initState);
};
