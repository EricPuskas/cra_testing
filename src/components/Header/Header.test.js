import React from "react";
import { shallow } from "enzyme";
import { findByAttribute } from "../../../utils/index";
import Header from "./Header";

describe("Header Component", () => {
  const component = shallow(<Header />);
  const wrapper = findByAttribute(component, "header");
  const logo = findByAttribute(component, "logo");

  it("Should render without errors", () => {
    expect(wrapper.length).toBe(1);
  });

  it("Should render a logo", () => {
    expect(logo.length).toBe(1);
  });

  it("Should pass snapshot test", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
