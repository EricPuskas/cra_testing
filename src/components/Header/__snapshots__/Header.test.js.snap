// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Should pass snapshot test 1`] = `"<header data-test=\\"header\\"><div class=\\"wrap\\"><div class=\\"logo\\"><img src=\\"logo.png\\" alt=\\"logo\\" data-test=\\"logo\\"/></div><div class=\\"logo-title\\">CRA-Testing</div></div></header>"`;
